src/server/server.ts

```diff
@@ -113,18 +115,49 @@ app.post('/transcripts/:studentID/:course', (req, res) => {
 // returns the student's grade in the specified course.
 // Fails if student or course is missing.
 // uses function getGrade(studentID:StudentID, course:Course) : number
-app.get('/transcripts/:studentID/:course', (req, res) => {
+app.get("/transcripts/:studentID/:course", (req, res) => {
   try {
     const studentID = parseInt(req.params.studentID);
-    const {course} = req.params;
+    const { course } = req.params;
     console.log(`Handling GET studentID=${studentID} course = ${course}`);
     const grade = db.getGrade(studentID, course);
     if (grade == undefined) {
-      res.status(400).send(`not grade for ${studentID} in ${course}`)
+      res.status(400).send(`not grade for ${studentID} in ${course}`);
+    }
+    res.status(200).send({ studentID, course, grade });
+  } catch (e) {
+    res.status(400).send(e);
+  }
+});
+
+app.get("/courses", (req, res) => {
+  try {
+    console.log("getting courses list with discription");
+    const courses = db.getCourseDescriptions();
+    res.status(200).send(courses);
+  } catch (e) {
+    res.status(400).send(e);
+  }
+});
+
+app.post("/courses", (req, res) => {
+  try {
+    console.log("posting a new course");
+  } catch (e) {
+    res.status(400).send(e);
+  }
+});
+
+app.get("/courses/:course", (req, res) => {
+  try {
+    console.log("getting a course with name");
+    const courseName = req.params.course
+    const course = db.getCourseDescription(courseName)
+    if (course === undefined) {
+      throw new Error(`Cannot find course with this name ${courseName}`);
```

Comment:
Can you enums or constants to store error messages and use those constants instead of hardcoding error messages.